apis/network/v1beta1/vpc_types.go

```diff
@@ -0,0 +1,157 @@
+
+	// The IPv4 CIDR block.
+	CIDRBlock *string `json:"cidrBlock,omitempty"`
+
+	// Information about the state of the CIDR block.
+	CIDRBlockState *VPCCIDRBlockState `json:"cidrBlockState,omitempty"`
+}
+
+// VPCIPv6CidrBlockAssociation represents the association of IPv6 CIDR blocks with the VPC.
+type VPCIPv6CidrBlockAssociation struct {
+
+	// The association ID for the IPv6 CIDR block.
+	AssociationID *string `json:"associationId,omitempty"`
+
+	// The IPv6 CIDR block.
+	IPv6CIDRBlock *string `json:"ipv6CidrBlock,omitempty"`
+
+	// Information about the state of the CIDR block.
+	IPv6CIDRBlockState *VPCCIDRBlockState `json:"ipv6CidrBlockState,omitempty"`
+
+	// The ID of the IPv6 address pool from which the IPv6 CIDR block is allocated.
+	IPv6Pool *string `json:"ipv6Pool,omitempty"`
+
+	// The name of the location from which we advertise the IPV6 CIDR block.
+	NetworkBorderGroup *string `json:"networkBorderGroup,omitempty"`
+}
+
+// VPCParameters define the desired state of an AWS Virtual Private Cloud.
+type VPCParameters struct {
+	// CIDRBlock is the IPv4 network range for the VPC, in CIDR notation. For
+	// example, 10.0.0.0/16.
+	// +kubebuilder:validation:Required
+	// +immutable
+	CIDRBlock string `json:"cidrBlock"`
+
+	// A boolean flag to enable/disable DNS support in the VPC
+	// +optional
+	EnableDNSSupport *bool `json:"enableDnsSupport,omitempty"`
+
+	// Tags are used as identification helpers between AWS resources.
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+
+	// Indicates whether the instances launched in the VPC get DNS hostnames.
+	// +optional
+	EnableDNSHostNames *bool `json:"enableDnsHostNames,omitempty"`
+
+	// The allowed tenancy of instances launched into the VPC.
+	// +optional
+	InstanceTenancy *string `json:"instanceTenancy,omitempty"`
+}
+
+// A VPCSpec defines the desired state of a VPC.
+type VPCSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  VPCParameters `json:"forProvider"`
+}
+
+// VPCObservation keeps the state for the external resource
+type VPCObservation struct {
+	// Information about the IPv4 CIDR blocks associated with the VPC.
+	CIDRBlockAssociationSet []VPCCIDRBlockAssociation `json:"cidrBlockAssociationSet,omitempty"`
+
+	// The ID of the set of DHCP options you've associated with the VPC.
+	DHCPOptionsID *string `json:"dhcpOptionsId,omitempty"`
+
+	// Information about the IPv6 CIDR blocks associated with the VPC.
+	IPv6CIDRBlockAssociationSet []VPCIPv6CidrBlockAssociation `json:"ipv6CidrBlockAssociationSet,omitempty"`
+
+	// Indicates whether the VPC is the default VPC.
+	IsDefault bool `json:"isDefault,omitempty"`
+
+	// The ID of the AWS account that owns the VPC.
+	OwnerID string `json:"ownerID,omitempty"`
+
+	// VPCState is the current state of the VPC.
+	VPCState string `json:"vpcState,omitempty"`
+
+	// Tags represents to current ec2 tags.
+	Tags []Tag `json:"tags,omitempty"`
+
+	// The ID of the VPC
+	VPCID string `json:"vpcId,omitempty"`
```

Comment:
Tags exist in spec and vpcid is the external-name, so, you can remove these from observation.